### ISSUE TITLE
Стандарты

### ISSUE THREAD

**Issue by maxsssssssss** [extremecodetv/ExtremeCodeOS#114]
Предлагаю внести стандарты в разработку:
1 Ограничение количество используемых в разработке языков до 4 (C#, C, C++, F#) (Для облегчения дополнения компилятора).
2 При добавление новых программ создавать или редактировать readme.md

Надеюсь идея хоть кому-то понравится.

**Comment by Nikita-bunikido** [comment]
Круто!
Писать на 10*e^340 япов писать не легко.
Про readme тож +, я и так их редактирую

**Comment by pheianox** [comment]
Не зря счетчик языков стоит в README.md ;)

**Comment by ketsushiri** [comment]
Если человек пишет что-то, что будет поддерживать только он сам, и что не будет затрагивать другие части проекта. Какую-нибудь софтину "в-себе", например, игру. то пускай хоть на лиспе пишет, имхо. Но вот на счёт системных или ещё каких-либо софтин, что явно или не явно перекликаются между собой и нуждаются в поддержке несколькими людьми, то регламентировать всё же следует.

**Comment by maxsssssssss** [comment]
@ketsushiri Тут видишь какая проблем. Если он и пишет программу для себя, то уже дополнять компилятор придется мне. Так как я не где не написал как добавить язык в компилятор.

**Comment by ketsushiri** [comment]
> Если он и пишет программу для себя, то уже дополнять компилятор придется мне.

@maxsssssssss Да, поэтому я предлагаю, чтобы при таких условиях, когда человек пишет программу на чём либо отличном от "стандартных" языков, компилировал бы он её сам и предоставлял программу уже с собранным бинарником. Было бы не очень правильно ограничевать людей, которые хотели бы помочь, но не могут.

**Comment by pheianox** [comment]
Если папку Compiler переместить в src проблем не возникнет?

**Comment by maxsssssssss** [comment]
@pheianox Возникнут! Компилятор это отдельная программа от OS из а чего сохранять её в исходниках OS будет нелогично! Но если отредактировать файл ComilerTypes.json то все будет работать.

**Comment by maxsssssssss** [comment]
@ketsushiri Ты прав и то же самое про интерпретируемые языки которые будут выполнятся во время работы OS. Ладно отредачу Проблему.
